### PR TITLE
build(docs): Enable dynamic wildcard redirects on FF.com

### PR DESCRIPTION
**Pull request by RishhiB**
The PR enables wildcard redirects in FF.com. When a user goes to a link that starts with `/docs/apis/*` it should redirect them to `/docs/api/<current-version>/*`. `/docs/api/current/*` and `/docs/apis/lts/*` are also included in this PR

## Instructions to reproduce the redirects;
1. `cd docs/ && pnpm i && pnpm run build && npm i -g @azure/static-web-apps-cli`
1. `swa start public --api-location api`
1. Modify the url from `/docs/api/<current-version>/` to `/docs/apis/` and see it go back to `/docs/api/<current-version>/`

### CODE REVIEW

**Comment by RishhiB** [comment]
Do i need docs/api/.funcignore, docs/api/.gitignore, docs/api/fallback/function.json, docs/api/fallback/sample.dat or docs/api/host.json?

**Comment by Josmithr** [comment]
> When a user goes to a link that starts with `/docs/apis/*` it should redirect them to `/docs/api/v1/*`

`apis` should redirect to `/docs/api/<current-version>`

**Comment by RishhiB** [comment]
> > When a user goes to a link that starts with `/docs/apis/*` it should redirect them to `/docs/api/v1/*`
> 
> `apis` should redirect to `/docs/api/<current-version>`

I have updated the PR's description

File: docs/api/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (Josmithr) [inline docs/api/package.json:1]: Does this directory need its own package.json? Can we manage the scripts in the `docs` root file instead?

**Review by Josmithr — COMMENTED** [review]
This looks good to me. I left a few comments regarding organization and documentation, but otherwise this looks good. That said, it's hard to evaluate without being able to test its integration with the production site. 

This shouldn't interfere with any other site functionality right? I.e. the worst thing that would happen if we checked this in is that the new redirects don't work correctly and we have to investigate? If so, maybe we can proceed that way? Otherwise, we probably need to find a way to publish a test version of the site, which I suspect would be non-trivial. @tylerbutler what do you think?

File: docs/api/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	"homepage": "https://fluidframework.com",
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/microsoft/FluidFramework.git",
+		"directory": "docs/api"
+	},
+	"license": "MIT",
+	"author": "Microsoft and contributors",
+	"scripts": {
+		"start": "func start",
```
Comment (Josmithr) [inline docs/api/package.json:15]: We should be sure to add the appropriate script for this to `docs`' `start` script so devs don't have to run more than one script to start the site.

**Comment by Josmithr** [comment]
> ... && pnpm install -g @azure/static-web-apps-cli`

Does this need to be installed globally? Or could we add it as a dev dep of the docs dir?

**Comment by Josmithr** [comment]
I pulled your branch and tried running everything locally. My first attempt to navigate to `/docs/apis/fluid-framework` attempted to redirect to `/docs/api/fluid-framework` (without the `v2` injected), which failed. Subsequent attempts to navigate to `/docs/apis/*` didn't result in any redirect attempts. I didn't see anything in the swa console to indicate why.

**Comment by RishhiB** [comment]
> This looks good to me. I left a few comments regarding organization and documentation, but otherwise this looks good. That said, it's hard to evaluate without being able to test its integration with the production site.
> 
> This shouldn't interfere with any other site functionality right? I.e. the worst thing that would happen if we checked this in is that the new redirects don't work correctly and we have to investigate? If so, maybe we can proceed that way? Otherwise, we probably need to find a way to publish a test version of the site, which I suspect would be non-trivial. @tylerbutler what do you think?

You are correct, the only issue I can think of is its not connected properly and no redirects happen. I dont think it can take down the site in production

**Comment by RishhiB** [comment]
I have encorporated the logic of docs/build-redirects.js inside the wildcard redirection logic, deleted the file and removed the npm script. I also was able to get the function to work without the package.json so I have removed that too

**Comment by RishhiB** [comment]
> > ... && pnpm install -g @azure/static-web-apps-cli`
> 
> Does this need to be installed globally? Or could we add it as a dev dep of the docs dir?

I have tried doing it as a dev dep, and it does work sometimes but when I push the change to the package.json and do a fresh install it did not work. consistently installing the swa cli using npm -g has been working. advising devs use that when testing

File: docs/api/package.json (hunk 1 above)
Comment (RishhiB) [reply docs/api/package.json]: Deleted it and it didnt affect local

File: docs/api/package.json (hunk 2 above)
Comment (RishhiB) [reply docs/api/package.json]: Deleted this file

**Comment by RishhiB** [comment]
> We can link to that from the root `docs` README

How can we can link docs/api/README.md from the root docs README?

**Review by Josmithr — COMMENTED** [review]
Based on our discussion, the only things I see blocking this are:
* Updating the local build workflow docs in the README.
* Ensuring devs can run a local build of the site via `pnpm i` => `npm run build` => `npm start` (no extra steps if we can help it)
* Ensure that Hugo's watch functionality still works while we're running swa

**Review by Josmithr — APPROVED** [review]
Looks good to me! Presumably a subsequent PR will enable redirects while running the site locally (once the SWA bug has been fixed)? 

If there are any relevant notes that should be added to the docs README, please add them. Otherwise 🚀

**Pull request merged**